api/v1alpha1/modelregistry_types.go

```diff
@@ -173,6 +173,209 @@ type GrpcSpec struct {
 	Image string `json:"image,omitempty"`
 }
 
+type ServerTLSSettings struct {
```

Comment:
basically, user can config their self-signed cert from DSCI, or just enable to use cluster default cert.
then ODH operator does create configmap with data, this configmap is created in every project (except k8s reserved ones or the one user excluded by annotation)
then it is up to each component to have logic consuming these configmap's data.

Reply:
@danielezonca the config map injected from platform is for a CA cert, afaik. To use the CA cert, we'd have to generate cert requests and then certs from that for model registry TLS endpoints. There are several parameters in cert creation (e.g. life of the cert to start) that user's should choose based on their security requirements. 
IMHO, model registry operator is not the right place to host cert creation logic. Cert management/rotation, etc. ideally should be done externally through a cert manager or a user selected solution like `LetsEncrypt`. 

In addition, servicemesh team wants to keep gateway config options in the mesh feature in odh operator. So, I'm working on removing most of the non-MR essential TLS settings from the MR CR. I'm not sure about the implementation details around it, since istio `virtualservice` is highly coupled with istio `gateway`. So, we'll need to work that out with the servicemesh team next week. 
If gateway resource is completely moved to the mesh feature in odh operator, MR CR won't have any TLS specific options, so the cert management won't be a model registry operator issue anymore. So, that's a nice thing! :smile:

Reply:
> model registry operator is not the right place to host cert creation logic. Cert management/rotation, etc. ideally should be done externally through a cert manager or a user selected solution like `LetsEncrypt`.

Totally agree, my comment is indeed to remove this configuration / management from MR API and leverage RHOAI operator, my expectation is that the integration with gateway is similar to what has been done with KServe.
@bartoszmajsak Feel free to jump in the conversation here :)

Reply:
Discussed with ossm team to go ahead with this stripped version of the CRD with only a secret name for TLS certs for now. Will refactor it later after Servicemesh integration for Gateway configuration is more concrete.